todo_app/app.py

```diff
@@ -42,43 +39,33 @@ def add_todo():
 
             to_add = {
                 'name': name,
+                'modified_date': datetime.now().isoformat(),
+                'is_done': False
             }
 
             if description.strip() != '':
-                to_add['desc'] = description
+                to_add['description'] = description
             
             if date.strip() != '':
-                to_add['due'] = datetime.strptime(date, '%d/%m/%Y').isoformat()
+                to_add['due_date'] = datetime.strptime(date, '%d/%m/%Y').isoformat()
             
-
-            trello = Trello()
-            trello.add_item(to_add)
+            db = DB()
+            db.add_item(MongoItem.from_dict(to_add, mode = "Save"))
         return redirect('/')
 
     @app.route('/todo/change-status/<id>', methods=['POST'])
     def change_todo_status(id):
-        credentials = get_trello_credentials()
-        trello = Trello()
-        todo_item = trello.get_item(id)
-        current_list = todo_item['idList']
-
-        new_list = None
-
-        if current_list == credentials['todo_list']:
-            new_list = credentials['completed_list']
-        else:
-            new_list = credentials['todo_list']
-
-        to_update = {
-            'idList': new_list
-        }
-        trello.update_item(id, to_update)
+        db = DB()
+        item_to_update = db.get_item(id)
+        item_to_update.update_status()
+        db.update_item(item_to_update)
         return redirect('/')
 
     @app.route('/todo/delete/<id>', methods=['POST'])
     def delete_todo(id):
-        trello = Trello()
-        trello.delete_item(id)
+        db = DB()
+        item_to_delete = db.get_item(id)
+        db.delete_item(item_to_delete)
```

Comment:
Here you make a call to the DB to load an item into memory based on its id, then you read the id back off the item in order to decide what to delete. Would having the `delete_item` method just take the id simplify this considerably?

Reply:
I did think about doing this. My thought process was to verify that it exists first but I suppose I am not actually checking it exists before passing the object to the delete_item function.

Reply:
Ah, yes I've seen that pattern before - your call, but I'd personally take the line that if it doesn't affect what you do then we might as well just let Mongo ignore the deletion if nothing matches 🙂